File: LeetCode/Easy/0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String cleaned = s.toLowerCase().replaceAll("[^a-z0-9]", "");
        return checkPalindrome(0, cleaned);
    }
    public boolean checkPalindrome( int i , String cleaned){
        int n = cleaned.length();
        if(i>=n/2)
        return true;
        if(cleaned.charAt(i) != cleaned.charAt(n-i-1))
    return false;
    return checkPalindrome(i+1, cleaned);
    }
}

File: LeetCode/Easy/0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isAlphaNum(char ch) {
        if ((ch >= '0' && ch <= '9') || 
            (tolower(ch) >= 'a' && tolower(ch) <= 'z')) {
            return true;
        }
        return false;
    }

    bool isPalindrome(string s) {
        int st =0;
        int end = s.length()-1;
        while(st<end){
            if(!isAlphaNum(s[st])){ //if char is not alphaNum skip it
                st++;
                continue;
            }
            if(!isAlphaNum(s[end])){
                end--;
                continue;
            }
            if(tolower(s[st])!=tolower(s[end])){
                return false;
            }
            else{
                st++;
                end--;
            }
        }
          return true;
    }
};